Remotion/Validation/UnitTests/Rules/PropertyValidationRuleTest.cs

```diff
@@ -1,30 +1,194 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using Moq;
+using NUnit.Framework;
+using Remotion.Reflection;
+using Remotion.Validation.Results;
 using Remotion.Validation.Rules;
 using Remotion.Validation.UnitTests.TestDomain;
+using Remotion.Validation.Validators;
 
 namespace Remotion.Validation.UnitTests.Rules
 {
   [TestFixture]
   public class PropertyValidationRuleTest
   {
     [Test]
-    [Ignore("TODO RM-5906")]
-    public void Validate ()
+    public void Validate_WithNullAsInstanceToValidate_ReturnsEmptyResult ()
     {
+      var context = new ValidationContext(null);
+
+      var validatorMock = new Mock<IPropertyValidator>();
+      validatorMock
+          .Setup(_ => _.Validate(It.IsAny<PropertyValidatorContext>()))
+          .Returns(Array.Empty<PropertyValidationFailure>());
+
+      var propertyInformationMock = new Mock<IPropertyInformation>();
+
+      Func<object, object> propertyFunc = i => i;
+      Func<object, bool> conditionFunc = i => i != null;
+
+      var rule = new PropertyValidationRule<object, int>(propertyInformationMock.Object, propertyFunc, conditionFunc, new[] { validatorMock.Object });
+
+      var res = rule.Validate(context);
+
+      Assert.That(res, Is.Empty);
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
-    public void IsActive ()
+    public void Validate_WithConditionIsFalse_ReturnsEmptyResult ()
     {
-      // consider better name than "IsActive"
+      var context = new ValidationContext("test");
+
+      var validatorMock = new Mock<IPropertyValidator>();
+      validatorMock
+          .Setup(_ => _.Validate(It.IsAny<PropertyValidatorContext>()))
+          .Returns(Array.Empty<PropertyValidationFailure>());
+
+      var propertyInformationMock = new Mock<IPropertyInformation>();
+
+      Func<object, object> propertyFunc = i => i;
+      Func<object, bool> conditionFunc = _ => false;
+
+      var rule = new PropertyValidationRule<object, int>(propertyInformationMock.Object, propertyFunc, conditionFunc, new[] { validatorMock.Object });
+
+      var res = rule.Validate(context);
+
+      Assert.That(res, Is.Empty);
+    }
+
+    [Test]
+    public void Validate_ReturnsListOfValidationFailures ()
+    {
+      var context = new ValidationContext("test");
+      var propertyInformationMock = new Mock<IPropertyInformation>();
+
+      var propertyFailure = new PropertyValidationFailure("test", propertyInformationMock.Object, "propval", "err", "validmessage");
+
+      var validatorMock = new Mock<IPropertyValidator>();
+      validatorMock
+          .Setup(_ => _.Validate(It.IsAny<PropertyValidatorContext>()))
+          .Returns(new[] { propertyFailure })
+          .Verifiable();
```

Comment:
We don't need this verifiable. If Validate isn't called, the result assertion wouldn't match.